src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/SymbolIsBannedInAnalyzersAnalyzer.cs

```diff
@@ -0,0 +1,123 @@
+            customTags: WellKnownDiagnosticTagsExtensions.Telemetry);
+
+        public static readonly DiagnosticDescriptor NoSettingSpecifiedSymbolIsBannedRule = new(
+            id: DiagnosticIds.NoSettingSpecifiedSymbolIsBannedInAnalyzersRuleId,
+            title: CreateLocalizableResourceString(nameof(NoSettingSpecifiedSymbolIsBannedInAnalyzersTitle)),
+            messageFormat: CreateLocalizableResourceString(nameof(NoSettingSpecifiedSymbolIsBannedInAnalyzersMessage)),
+            category: DiagnosticCategory.MicrosoftCodeAnalysisCorrectness,
+            defaultSeverity: DiagnosticSeverity.Error,
+            isEnabledByDefault: true,
+            description: CreateLocalizableResourceString(nameof(NoSettingSpecifiedSymbolIsBannedInAnalyzersDescription)),
+            helpLinkUri: null,
+            customTags: WellKnownDiagnosticTagsExtensions.Telemetry);
+    }
+
+    public abstract class SymbolIsBannedInAnalyzersAnalyzer<TSyntaxKind> : SymbolIsBannedAnalyzerBase<TSyntaxKind>
+        where TSyntaxKind : struct
+    {
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(SymbolIsBannedInAnalyzersAnalyzer.SymbolIsBannedRule, SymbolIsBannedInAnalyzersAnalyzer.NoSettingSpecifiedSymbolIsBannedRule);
+
+        protected sealed override bool OptedInToBannedSymbolEnforcement(CompilationStartAnalysisContext compilationContext, SyntaxNode syntax)
+        {
+            return compilationContext.Options.GetBoolOptionValue(
+                EditorConfigOptionNames.EnforceExtendedAnalyzerRules,
+                rule: null,
+                syntax.SyntaxTree,
+                compilationContext.Compilation,
+                defaultValue: false);
+        }
+
+        protected sealed override DiagnosticDescriptor SymbolIsBannedRule { get; } = SymbolIsBannedInAnalyzersAnalyzer.SymbolIsBannedRule;
+
+#pragma warning disable RS1012 // 'compilationContext' does not register any analyzer actions. Consider moving actions registered in 'Initialize' that depend on this start action to 'compilationContext'.
+        protected sealed override Dictionary<ISymbol, BanFileEntry>? ReadBannedApis(CompilationStartAnalysisContext compilationContext)
+        {
+            var provider = WellKnownTypeProvider.GetOrCreate(compilationContext.Compilation);
+            var diagnosticAnalyzerAttributeType = provider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisDiagnosticsDiagnosticAnalyzerAttribute);
+            var generatorAttributeType = provider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute);
+            compilationContext.RegisterSymbolAction(analyzePossibleAnalyzerOrGenerator, SymbolKind.NamedType);
+
+            const string fileName = "Microsoft.CodeAnalysis.Analyzers.AnalyzerBannedSymbols.txt";
+            var stream = typeof(SymbolIsBannedInAnalyzersAnalyzer<>).Assembly.GetManifestResourceStream(fileName);
```

Comment:
Another option is to package this text file as an additional file in the NuGet package, and have the analyzer access it via AdditionalFiles on the analysis context. This approach has the advantage that the user has the option to hand edit the txt file within the downloaded NuGet package on their machine to include/exclude additional banned APIs in it for customized experience.

In case you would like to take this route: We have post-build tooling in the repo that generates package specific targets file and embeds it into the NuGet package. You can add the `ItemGroup` to always add this txt file as an additional file by appending to the below string: https://github.com/dotnet/roslyn-analyzers/blob/d80e535c08bf3eace650005ac2e7e061ed8692cd/src/Tools/GenerateDocumentationAndConfigFiles/Program.cs#L1560-L1575

You would then need to update https://github.com/dotnet/roslyn-analyzers/blob/main/nuget/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.Package.csproj to include this file as an `AnalyzerNupkgFile`, similar to https://github.com/dotnet/roslyn-analyzers/blob/3393ecfccd4331ced6423e9b875187ee377f1ce2/nuget/PerformanceSensitiveAnalyzers/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers.Package.csproj#L22

Feel free to ignore this suggestion completely :-)

Reply:
Let's track this as a suggestion, and if we find that people need this in order to onboard to the new analyzer, then we'll add it at that point.